src/pybind/mgr/balancer/module.py

```diff
@@ -1217,7 +1217,7 @@ def do_crush_compat(self, plan: MsPlan) -> Tuple[int, str]:
                           pe.score)
             plan.compat_ws = {}
             return -errno.EDOM, 'Unable to find further optimization, ' \
-                                'change balancer mode and retry might help'
+                                'change balancer mode or adjust target_max_misplaced_ratio config and retry might help'
```

Comment:
Suggestion to make the wording flow a bit better:

```suggestion
                                'changing the balancer mode or adjusting the target_max_misplaced_ratio config, then retrying, might help'
```

Also, it might be worthwhile to add this scenario to the balancer QA test to make sure you're getting the desired result: https://github.com/ceph/ceph/blob/f0c8fd73a0190099a60ba56a8c2f6bdc28d8beef/qa/standalone/mgr/balancer.sh#L78-L86
Let me know if you'd feel comfortable adding this.